### PR TITLE
samples: nfc: Add support for nRF5340dk

### PR DESCRIPTION
**Pull request by KAGA164**
This PR adds support for nRF5340dk in NFC samples

### CODE REVIEW

**Review by piwe-nordic — APPROVED** [review]

File: samples/bluetooth/central_nfc_pairing/CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,9 @@ cmake_minimum_required(VERSION 3.13.1)
 set(NRF_SUPPORTED_BOARDS
   nrf52dk_nrf52832
   nrf52840dk_nrf52840
+  nrf52833dk_nrf52833
   nrf5340pdk_nrf5340_cpuapp
+  nrf5340dk_nrf5340_cpuapp
```
Comment (grochu) [inline samples/bluetooth/central_nfc_pairing/CMakeLists.txt:15]: Is there an issue with _ST25R3911B_ when building for `nrf5340dk_nrf5340_cpuappns`?

**Review by grochu — COMMENTED** [review]
Probably similar changes are required for BLE samples.

**Comment by grochu** [comment]
According to @carlescufi the PDK boards need to be removed from the _allow-list_

**Review by grochu — CHANGES_REQUESTED** [review]
Please remove the PDK from yaml files.

File: samples/bluetooth/peripheral_nfc_pairing/sample.yaml [hunk 2]
```diff
@@ -5,5 +5,6 @@ tests:
   samples.bluetooth.peripheral_nfc_pairing:
     build_only: true
     build_on_all: true
-    platform_allow: nrf52840_pca10056 nrf52_pca10040 nrf5340pdk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840_pca10056 nrf52_pca10040 nrf5340pdk_nrf5340_cpuapp nrf5340pdk_nrf5340_cpuappns
```
Comment (nvlsianpu) [inline samples/bluetooth/peripheral_nfc_pairing/sample.yaml:8]: nrf52840_pca10056 nrf52_pca10040 -> nrf52840dk_nrf52840 nrf52dk_nrf52832

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: samples/bluetooth/central_nfc_pairing/CMakeLists.txt (hunk 1 above)
Comment (KAGA164) [reply samples/bluetooth/central_nfc_pairing/CMakeLists.txt:15]: There isn't issue but the overlay files have to be prepared for `nrf5340dk_nrf5340_cpuappns`

File: samples/bluetooth/peripheral_nfc_pairing/sample.yaml (hunk 2 above)
Comment (KAGA164) [reply samples/bluetooth/peripheral_nfc_pairing/sample.yaml]: Thanks for noticing. Done

**Comment by KAGA164** [comment]
@grochu PDK target is removed from yaml files

**Review by grochu — APPROVED** [review]
Looks good.

**Review by nvlsianpu — APPROVED** [review]

**Comment by lemrey** [comment]
I suppose we should remove that from the `CMakeLists.txt` files too?

**Comment by KAGA164** [comment]
@lemrey I am not sure about removing this board from `CMakeLists.txt` then you won't be able to build these samples for the nrf5340pdk. Should we remove a support for nrf5340pdk in NFC samples ? @grochu @carlescufi

**Comment by grochu** [comment]
> @lemrey I am not sure about removing this board from `CMakeLists.txt` then you won't be able to build these samples for the nrf5340pdk. Should we remove a support for nrf5340pdk in NFC samples ? @grochu @carlescufi

I am of the opinion that the PDK should remain in Cmake, as some users might still only use PDKs, and I can't see the point in forbidding them doing this.

**Comment by thst-nordic** [comment]
this PR will fix the sanitycheck failures on master.  is it ready to merge once CI passes?

**Review by utzig — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Comment by lemrey** [comment]
Ok, was just wondering

**Comment by thst-nordic** [comment]
These changes only effect sanitycheck tests.   Those passed on CI, and we skipped the other system tests to save resources.

**Pull request merged**